tools/azure-pipelines/jobs-template.yml

```diff
@@ -129,6 +129,22 @@ jobs:
     condition: not(eq('${{parameters.test_pool_definition.name}}', 'Default'))
     displayName: Cache Maven local repo
 
+  - task: Cache@2
+    inputs:
+      key: '"$(module)" | $(TESTCONTAINER_CACHE_KEY)'
+      restoreKeys: '"$(module)" | $(TESTCONTAINER_CACHE_FALLBACK_KEY)'
+      path: $(TESTCONTAINER_CACHE_FOLDER)
+      cacheHitVar: TESTCONTAINER_CACHE_HIT
+    continueOnError: true
+    condition: not(eq('${{parameters.test_pool_definition.name}}', 'Default'))
+    displayName: Cache Testcontainer docker images
+
+  - script: |
+      docker load -i $(TESTCONTAINER_CACHE_FOLDER)/cache.tar || true
+    displayName: Restore Testcontainer docker images
+    condition: and(not(canceled()), eq(variables.TESTCONTAINER_CACHE_HIT, 'true'))
+    continueOnError: true
```

Comment:
Based on [this build](https://dev.azure.com/Autophagy/Flink-CI/_build/results?buildId=80&view=logs&j=9dc1b5dc-bcfa-5f83-eaa7-0cb181ddc267&t=b1670b3c-3f19-5f34-3e91-67ac7d80384f) this doesn't seem to work.

Reply:
Huh, curious. Previous tests worked fine: https://dev.azure.com/Autophagy/Flink-CI/_build/results?buildId=78&view=logs&j=fba17979-6d2e-591d-72f1-97cf42797c11&t=f19988ba-dc4e-5334-707d-a0c139178ea5

I'll recheck the condition.

Reply:
In that build it was also skipped for `core`, but it worked for the python one. That is indeed weird. https://dev.azure.com/Autophagy/Flink-CI/_build/results?buildId=78&view=logs&j=9dc1b5dc-bcfa-5f83-eaa7-0cb181ddc267&t=b1670b3c-3f19-5f34-3e91-67ac7d80384f

Reply:
Ah, I think I understand: there wasnt a cache hit for the key, so it fell to the fallback cache which populates `TEST_CONTAINER_CACHE_HIT` with `inexact` rather than `true`. Switching to a static cache key and removing the fallback key will mean this won't occur again. Thanks for pointing it out :sweat_smile: